core/src/main/java/pl/allegro/tech/embeddedelasticsearch/EmbeddedElastic.java

```diff
@@ -129,6 +130,31 @@ public void index(String indexName, String indexType, List<CharSequence> jsons)
         elasticRestClient.indexWithIds(indexName, indexType, jsons.stream().map(json -> new DocumentWithId(null, json.toString())).collect(Collectors.toList()));
     }
 
+    /**
+     * Index single document document
+     *
+     * @param indexName target index
+     * @param indexType target index name
+     * @param docId     document id
+     * @param json      document represented as JSON
+     */
+    public void index(String indexName, String indexType, String docId, String json) {
+        index(indexName, indexType, docId, null, json);
+    }
+
+    /**
+     * Index single document document with routing
+     *
+     * @param indexName target index
+     * @param indexType target index name
+     * @param docId     document id
+     * @param routing   routing
+     * @param json      document represented as JSON
+     */
+    public void index(String indexName, String indexType, String docId, String routing, String json) {
```

Comment:
We had 3 arguments in index method and I think that is the limit we should exceed :) For me this is the moment when we should create some IndexRequest (along with builder) class where it will be possible to specify those extra options